build.sc

```diff
@@ -1559,191 +1650,301 @@ object dev extends MillModule {
+      workingDir = workDir
+    )
+    PathRef(workDir / "build" / "site")
+  }
+  def source0: Source = T.source(millSourcePath)
+  def source = T{
+    os.list(source0().path).foreach(p => os.copy(p, T.dest / p.relativeTo(source0().path)))
+
+    val pagesWd = T.dest / "modules" / "ROOT" / "pages"
+
+    val renderedExamples: Seq[(os.SubPath, PathRef)] =
+      T.traverse(example.exampleModules)(m => T.task {
+        (m.millSourcePath.subRelativeTo(example.millSourcePath), m.rendered())
+      })()
+
+    for ((name, pref) <- renderedExamples) os.copy(
+      pref.path,
+      pagesWd / "example" / os.SubPath(s"$name.adoc"),
+      createFolders = true
+    )
+
+    val contribReadmes = T.traverse(contrib.contribModules)(m =>
+      T.task {
+        m.millModuleSegments.last.render -> m.readme()
+      }
+    )()
+
+    for ((name, pref) <- contribReadmes) os.copy(
+      pref.path,
+      pagesWd / "contrib" / s"${name}.adoc",
+      createFolders = true
+    )
+
+    PathRef(T.dest)
+  }
+  def supplementalFiles = T.source(millSourcePath / "supplemental-ui")
+  def devAntoraSources: Target[PathRef] = T {
+    val dest = T.dest
+    shared.mycopy(source().path, dest, mergeFolders = true)
+    val lines = os.read(dest / "antora.yml").linesIterator.map {
+      case l if l.startsWith("version:") =>
+        s"version: 'master'" + "\n" + s"display-version: '${millVersion()}'"
+      case l if l.startsWith("    mill-version:") =>
+        s"    mill-version: '${millVersion()}'"
+      case l if l.startsWith("    mill-last-tag:") =>
+        s"    mill-last-tag: '${millLastTag()}'"
+      case l => l
     }
-    def generatePages(authorMode: Boolean) = T.task {
-      // dependency to sources
-      source()
-      val docSite = T.dest
-      val playbook = docSite / "antora-playbook.yml"
-      val siteDir = docSite / "site"
-      os.write(
-        target = playbook,
-        data = githubPagesPlaybookText(authorMode)(),
-        createFolders = true
-      )
-      // check xrefs
-      runAntora(
-        npmDir = npmBase(),
-        workDir = docSite,
-        args = Seq(
-          "--generator",
-          "@antora/xref-validator",
-          playbook.last,
-          "--to-dir",
-          siteDir.toString(),
-          "--attribute",
-          "page-pagination"
-        ) ++
-          Seq("--fetch").filter(_ => !authorMode)
-      )
-      // generate site (we can skip the --fetch now)
-      runAntora(
-        npmDir = npmBase(),
-        workDir = docSite,
-        args = Seq(
-          playbook.last,
-          "--to-dir",
-          siteDir.toString(),
-          "--attribute",
-          "page-pagination"
-        )
+    os.write.over(dest / "antora.yml", lines.mkString("\n"))
+    PathRef(dest)
+  }
+  def githubPagesPlaybookText(authorMode: Boolean): Task[String] = T.task {
+    s"""site:
+       |  title: Mill
+       |  url: ${if (authorMode) s"${T.dest}/site/index.html" else Settings.docUrl}
```

Comment:
This makes the home navigation links in the top left work even when looking at the local version of the site